### ISSUE TITLE
Investigate Android Game Browser ANR (Application Not Responding)

### ISSUE THREAD

**Issue by carstene1ns** [EasyRPG/Player#1330]
These are happening quite often in the current version (may be an issue of the updated Android ndk/sdk/build tools):
**Input dispatching timed out (Waiting because no window has focus but there is a focused application that may eventually add a window when it finishes starting up.)**

```
"main" tid=1 Native
"main" prio=5 tid=1 Native
  | group="main" sCount=1 dsCount=0 obj=0x746264b0 self=0xf4f36500
  | sysTid=27481 nice=0 cgrp=default sched=0/0 handle=0xf71c5b34
  | state=S schedstat=( 4707946272 249162681 25230 ) utm=144 stm=326 core=5 HZ=100
  | stack=0xff1f7000-0xff1f9000 stackSize=8MB
  | held mutexes=

   #00  pc 0000000000040f40  /system/lib/libc.so (__getdents64+12)
   #01  pc 000000000001a0c5  /system/lib/libc.so (_ZL16__readdir_lockedP3DIR+20)
   #02  pc 000000000001a19f  /system/lib/libc.so (readdir+16)
   #03  pc 00000000000141e7  /system/lib/libjavacore.so (???)
   #04  pc 000000000002ddf1  /data/dalvik-cache/arm/system@framework@boot.oat (Java_java_io_File_listImpl__Ljava_lang_String_2+92)
   at java.io.File.listImpl (Native method)
   at java.io.File.list (File.java:740)
   at java.io.File.listFiles (File.java:782)

   at org.easyrpg.player.game_browser.GameScanner.scanFolder (GameScanner.java:144)
   at org.easyrpg.player.game_browser.GameScanner.scanGames (GameScanner.java:92)
   at org.easyrpg.player.game_browser.GameScanner.getInstance (GameScanner.java:51)
   at org.easyrpg.player.game_browser.GameBrowserActivity.displayGameList (GameBrowserActivity.java:161)
   at org.easyrpg.player.game_browser.GameBrowserActivity.onResume (GameBrowserActivity.java:92)

   at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1259)
   at android.app.Activity.performResume (Activity.java:6399)
   at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3110)
   at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3152)
   at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2499)
   at android.app.ActivityThread.access$900 (ActivityThread.java:153)
   at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1347)
   at android.os.Handler.dispatchMessage (Handler.java:102)
   at android.os.Looper.loop (Looper.java:148)
   at android.app.ActivityThread.main (ActivityThread.java:5451)
   at java.lang.reflect.Method.invoke! (Native method)
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:756)
   at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:646)
```

**Comment by Ghabry** [comment]
That happens because the game search by the game browser runs in the UI thread.

**Comment by carstene1ns** [comment]
I guess we may be hitting some timeout sometimes then.

Some recent rating:
> I can't select the game folder because when I click on it the application freezes.

**Comment by Ghabry** [comment]
This should be fixed by #2699 
The Game browser locates the games now in some background thing. (No idea how BlisterB implemented it but it does not hang anymore :D)

**Comment by BlisterB** [comment]
Yes now the app display a loading logo while scanning in the background :)
